docs/reference/filebeat/file-identity.md

```diff
@@ -0,0 +1,183 @@
+```
+
+As you can see, on APFS all three files have different inode values: 112076744, 112076745, and 112076746. So, this works as expected and creates stable identifiers.
+
+However, if you run the same script in an Ubuntu Docker container:
+
+```
+inode-test created with inode '1715023 inode-test'
+Copied inode-test->inode-test-copy, the new inode for the copy '1715026 inode-test-copy'
+inode-test has been deleted
+ls: cannot access 'inode-test': No such file or directory
+After copying inode-test-copy back to inode-test the inode is '1715023 inode-test'
+```
+
+You can see that the file system cached the inode value from the first file, which we deleted, and reused it for the second copy with the same filename: 1715023, 1715026, and 1715023.
+
+Even worse, it does not even have to be the same filename; a different file can reuse the same inode:
+
+```sh
+# touch x
+# ls -i x
+1715023 x # inode value for the first file
+# rm x
+# touch y
+# ls -i y
+1715023 y # same inode value for the second file with a different name
+```
+
+As mentioned before, it's common for containerized environments and network shares. If you're planning to run Filebeat in such an environment, avoid using this file identity.
+
+It's not only `inode` values: when using the Linux [LVM](https://en.wikipedia.org/wiki/Logical_Volume_Manager_%28Linux%29) (Logical Volume Manager), device numbers are allocated dynamically at module load (refer to [Persistent Device Numbers](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/logical_volume_manager_administration/lv#persistent_numbers) in the Red Hat Enterprise Linux documentation).
+
+However, this problem with device numbers can be fixed by our next file identity option.
+
+### `inode_marker`
+
+Let's say you verified that your file system does not re-use `inode` values and it's safe to rely on them. But you found out that the device number was changing and caused Filebeat to re-ingest all the data from the beginning causing data duplication.
+
+This can be fixed by using a special marker file on your file system instead of asking the OS for a device number.
+
+Users specify a path by setting `file_identity.inode_marker.path`. This path leads to a file that has some short unique text that remains unchanged forever. The content of this file replaces the `device` number in the `native` file identity. The rest remains the same as `native`.
+
+### `fingerprint`
+
+The most stable file identity implementation at the moment. This file identity is default since Filebeat version 9.0.
+
+This file identity is using file fingerprints produced by the scanner component of the filestream input. Therefore, in order to activate this file identity, it's required to first activate fingerprint in the scanner by setting `prospector.scanner.fingerprint.enabled: true`.
+
+In the context of the filestream input, fingerprint is a hash string computed from a part of the file content.
```

Comment:
I'd add here the hash algorithm used instead of just "hash".

Reply:
It's not just "hash" it' a "hash string" here which I believe is clear.